### ISSUE TITLE
Lack of limitations/clarity

### ISSUE THREAD

**Issue by ruippgoncalves** [collimarco/Standard-Source-Available-License#1]
Being honest I think that this license lacks some limitations/clarity.

Although the redistribution should be a licensor option (but, generally also applied with these kinds of licenses, I suppose that this could be an optional limitation making this license more modular), the purpose of shared source is to the copyright owner be able to sell the software. But in case of a software that can be used in a SaaS manner, support/consulting, and other circumstances, nothing is stated.

For example, the Commons Clause specifically states the following:

    ... the grant of rights under the License will not include, and the License does not grant to you, the right to Sell the Software.

    For purposes of the foregoing, “Sell” means practicing any or all of the rights granted to you under the License to provide to third parties, for a fee or other consideration (including without limitation fees for hosting or consulting/ support services related to the Software), a product or service whose value derives, entirely or substantially, from the functionality of the Software. Any license notice or attribution required by the License must also include this Commons Clause License Condition notice.

Another thing that isn't clear is that in case of the copyright owner selling licenses, can they be resold or transferred? (But I guess that this may not be in the software license itself, but in the license purchased). Nevertheless, I just wanted to point that out.

Also, I think that the license should be more explicit in some points, like saying that "other methods to circumvent the license verification aren't allowed" (I'm talking about keygens, as it is not a modification to the source).

At last, if you opt for using the modularity, you should also make the point 2 optional, as a lot of software may just want to prevent other from "selling" (as in the commons clause definition) the software. (I have a project of mine that does not has license key (it's free) but I do not want others to make profit out of him).

Of course, this is just my opinion in this license, but has I use a "forked" version of this license, because of the motives explained above, I think it may be useful for other people as well. What are your thoughts this?

**Comment by ruippgoncalves** [comment]
As an example of what I purpose:
ruippgoncalves/Standard-Source-Available-License@1fcbdf73bb471fe884461966d4169cd583d2859c
If you want I can send it as a pull request.

**Comment by collimarco** [comment]
Hello @ruippgoncalves ,

Thanks for your contributions!

> the purpose of shared source is to the copyright owner be able to sell the software. But in case of a software that can be used in a SaaS manner, support/consulting, and other circumstances, nothing is stated.

The goal of this license (SSAL) is to be minimalist, but flexible about the business model.

Basically the legal terms (point 2) delegate to the executable code the details of the license mechanism and state that it's illegal to alter that mechanism.

In case of a SaaS, for example, you can include code in the software that checks the usage (requests, RAM, cluster size, etc.) and bills accordingly (or blocks the usage without a valid license key).

> the Commons Clause specifically states the following
> the grant of rights under the License will not include, and the License does not grant to you, the right to Sell the Software.

The SSAL license doesn't care about what you do with the software, you could even sell it (e.g. as part of a larger product). The only point is that you can't alter the code related the the validation of the license: in this way the people that buy the software, even indirectly, will have to purchase a license ALSO from the original author (otherwise the software doesn't work).

> "other methods to circumvent the license verification aren't allowed" (I'm talking about keygens, as it is not a modification to the source)

I'm assuming that the code uses JWT, cryptography or internet activation to make sure that only valid license keys are accepted.

**You can easily write a code that, if it's not altered, is 100% secure and cannot be broken.** 

Here's [a software](https://github.com/cuber-cloud/cuber-gem) that uses JWT for the license key. Unless you change the code directly (which is illegal) you cannot break the license validation.

> a lot of software may just want to prevent other from "selling" (as in the commons clause definition) the software

This is not aligned with the aim of this license. Most people want to make profits, so why prevent them from selling our software? Instead, we simply ask to pay us our part (i.e. the license key).

Basically the end-buyers will have to buy a license from the original author: he is the only one who is able to generate the license keys.

**Comment by collimarco** [comment]
In addition to the above considerations and about your concerns about other people "selling", note that currently this license allows "use and modifications".  It doesn't even allow "redistribution"/"selling" in the current version (and if it ever will, my above reasoning applies).
